modules/basic/ds/hashmap.h

```diff
@@ -217,6 +225,262 @@ class HashmapBuilder : public HashmapBaseBuilder<K, V, H, E> {
+    return V(0);
+  }
+
+  /**
+   * @brief Get the const mapping value of the given key.
+   *
+   */
+  const V& at(const K& key) const {
+    // return hashmap_.at(key);
+    LOG(INFO) << __func__ << " is not finished yet.";
+    return V(0);
+  }
+
+  /**
+   * @brief Get the size of the hashmap.
+   *
+   */
+  size_t size() const { return n_elements_; }
+
+  /**
+   * @brief Reserve the size for the hashmap.
+   *
+   */
+  void reserve(size_t size) {
+    LOG(INFO) << __func__ << " is not finished.";
+    vec_kv_.reserve(size);
+  }
+
+  /**
+   * @brief Return the maximum possible size of the HashMap, i.e., the number
+   * of elements that can be stored in the HashMap.
+   *
+   */
+  size_t bucket_count() const {
+    // return hashmap_.bucket_count();
+    LOG(INFO) << __func__ << " is not finished yet.";
+    return 0;
+  }
+
+  /**
+   * @brief Return the load factor of the HashMap.
+   *
+   */
+  float load_factor() const {
+    // return hashmap_.load_factor();
+    LOG(INFO) << __func__ << " is not finished yet.";
+    return 0;
+  }
+
+  /**
+   * @brief Check whether the hashmap is empty.
+   *
+   */
+  bool empty() const {
+    // return hashmap_.empty();
+    LOG(INFO) << __func__ << " is not finished yet.";
+    return true;
+  }
+
+  /**
+   * @brief Return the beginning iterator.
+   *
+   */
+  void* begin() {
+    // return hashmap_.begin();
+    LOG(INFO) << __func__ << " is not finished yet.";
+    return nullptr;
+  }
+
+  /**
+   * @brief Return the const beginning iterator.
+   *
+   */
+  void* begin() const {
+    // return hashmap_.begin();
+    LOG(INFO) << __func__ << " is not finished yet.";
+    return nullptr;
+  }
+
+  /**
+   * @brief Return the const beginning iterator.
+   *
+   */
+  void* cbegin() const {
+    // return hashmap_.cbegin();
+    LOG(INFO) << __func__ << " is not finished yet.";
+    return nullptr;
+  }
+
+  /**
+   * @brief Return the ending iterator
+   *
+   */
+  void* end() {
+    // return hashmap_.end();
+    LOG(INFO) << __func__ << " is not finished yet.";
+    return nullptr;
+  }
+
+  /**
+   * @brief Return the const ending iterator.
+   *
+   */
+  void* end() const {
+    // return hashmap_.end();
+    LOG(INFO) << __func__ << " is not finished yet.";
+    return nullptr;
+  }
+
+  /**
+   * @brief Return the const ending iterator.
+   *
+   */
+  void* cend() const {
+    // return hashmap_.cend();
+    LOG(INFO) << __func__ << " is not finished yet.";
+    return nullptr;
+  }
+
+  /**
+   * @brief Find the value by key.
+   *
+   */
+  void* find(const K& key) {
+    // return hashmap_.find(key);
+    LOG(INFO) << __func__ << " is not finished yet.";
+    return nullptr;
+  }
+
+  /**
+   * @brief Associated with a given data buffer
+   */
+  void AssociateDataBuffer(std::shared_ptr<Blob> data_buffer) {
+    // this->data_buffer_ = data_buffer;
+  }
+
+  template <typename K_ = K>
+  typename std::enable_if<std::is_integral<K_>::value, void>::type Construct(
```

Comment:
I suggest adding a static_assert to check if the `K` is integral in to the class, rather than check this method only.